src/transaction/txParser.ts

```diff
@@ -148,14 +160,37 @@ const parseRelay = (poolRelay: any): _PoolRelay => {
 }
 
 const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
+  const parseStakeCredentials = (stakeCredentials: any): StakeCredentials => {
+    if (Array.isArray(stakeCredentials) && stakeCredentials.length === 2) {
+      if (stakeCredentials[0] === StakeCredentialsKeys.ADDR_KEY_HASH) {
+        return {
+          type: StakeCredentialsKeys.ADDR_KEY_HASH,
+          addrKeyHash: stakeCredentials[1],
+        }
+      }
+      if (stakeCredentials[0] === StakeCredentialsKeys.SCRIPT_HASH) {
+        return {
+          type: StakeCredentialsKeys.SCRIPT_HASH,
+          scriptHash: stakeCredentials[1],
+        }
+      }
+    } else if (Buffer.isBuffer(stakeCredentials)) {
+      return {
+        type: StakeCredentialsKeys.ADDR_KEY_HASH,
+        addrKeyHash: stakeCredentials,
+      }
+    }
+    throw Error(Errors.StakeCredentialsParseError)
+  }
+
   const stakeKeyRegistrationCertParser = (
     txCertificate: any,
   ): _StakingKeyRegistrationCert => {
     if (!isTxStakingKeyRegistrationCert(txCertificate)) {
       throw Error(Errors.TxStakingKeyRegistrationCertParseError)
     }
-    const [type, [, pubKeyHash]] = txCertificate
-    return ({ type, pubKeyHash })
+    const [type, [, stakeCredentials]] = txCertificate
```

Comment:
I think this should rather be:
```suggestion
    const [type, stakeCredentials] = txCertificate
```
And similarly for the other certificates.